libs/hdf-converters/src/splunk-mapper.ts

```diff
@@ -152,42 +151,45 @@ function consolidateFilePayloads(
 }
 
 export async function checkSplunkCredentials(
-  config: SplunkConfig,
-  webCompatibility: boolean
-): Promise<boolean> {
-  let service: splunkjs.Service;
-  if (webCompatibility) {
-    service = new splunkjs.Service(new ProxyHTTP.JQueryHttp(''), config);
-  } else {
-    service = new splunkjs.Service(config);
-  }
+  config: SplunkConfig
+): Promise<string> {
+  const hostname = config.port
+    ? `${config.scheme}://${config.host}:${config.port}`
+    : `${config.scheme}://${config.host}:8089`;
+  const username = (config.username ??= '');
+  const password = (config.password ??= '');
```

Comment:
What does ??= do?

Reply:
[Nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment). Only assigns the username/password if it is null.

Reply:
Nice